### ISSUE TITLE
"This value is not a valid URL" on an URL from DigiKey

### ISSUE THREAD

**Issue by nomike** [Part-DB/Part-DB-server#521]
**Describe the bug**
When I try to scrape a specific part from DigiKey, Part-DB doesn't allow me to create it complaining about an invalid URL.

The error message is "This value is not a valid URL.".

The URL is the one of the main image:
https://mm.digikey.com/Volume0/opasdata/d220001/medias/images/1132/3302^24(300SF).jpg

I highly suspect this is because of the "^" character being present in the URL.

RFC 1738  defines this character as being unsafe and demands it being encoded as `%5E`. So I guess the issue is with DigiKey as they should not return an URL like that.

I raised a support ticket with DigiKey, maybe they can fix it on their end. I think adding a dodgy workaround to Part-DB is a particularly ugly way to solve this issue.

I was able to save the part once I manually changed the URL to properly encode the character, so there is at least a workaround for now, even though it is not very user friendly.

**To Reproduce**
1. Setup the DigiKey connector
2. Go to "Tools" → "Create parts from info provider"
3. Keyword "3M157903-1-ND" with provider "DigiKey" gives you one result which is a flat ribbon cable.
4. Click on the "+" icon to add create a new part.
5. Click on "Save changes"

**Expected behavior**
The part should should be created and saved to the database.

**Screenshots**

**Server Side**
- Part-DB Version: 1.10.7 
- PHP Version: 8.2.7 (SAPI: apache2handler)
- Database Server: sqlite 3.40.1

**Desktop (please complete the following information):**
 - OS: Ubuntu 23.10
 - Browser: Firefox
 - Version: 122.0.1 (64-bit)

**Additional context**
None

**Comment by jbtronics** [comment]
It seems that digikey has somehow fixed this in their backend in the meantime. However, I still implemented a quick replacement of invalid characters with escaped forms for all info providers, to prevent such issues in the future.

**Comment by nomike** [comment]
I've just updated to v. 1.11.1 today and just now ran into this issue again.

DigiKey Part Number: 1/2-18-8992
Issue is with Attachments, Main-Image.
URL from DigiKey: https://mm.digikey.com/Volume0/opasdata/d220001/medias/images/2524/1^2-18-8992.JPG

So first of all it looks like DigiKey didn't fix it on their end. They may have fixed that one URL, but that seems to be it.

Secondly: Your Workaround doesn't seem to work .Or is it not included in v.1.11.1?

Just a quick note, to rule out the issue being on me doing the upgrade incorrectly:

I've logged on to my server, and ran
`git fetch -p && git checkout v.1.11.1 && php bin/console cache:clear` to upgrade to the new version. It also displays as v.1.11.1 in the GUI now.
